### PR TITLE
Add source: Kiertokapula FI

### PR DESCRIPTION
**Pull request by developerfromjokela**
Adding source for Finnish waste utility company Kiertokapula, which operates in cities/towns: 

- HATTULA
- HAUSJÄRVI
- HYVINKÄÄ
- HÄMEENLINNA
- JANAKKALA
- JÄRVENPÄÄ
- KERAVA
- LOPPI
- MÄNTSÄLÄ
- NURMIJÄRVI
- RIIHIMÄKI
- SÄHAS TALOYHTIÖT
- TUUSULA
- VALKEAKOSKI

### CODE REVIEW

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        for estate in data.values():
+            for customer in estate:
+                r = session.get(API_URL + '/secure/get_services_by_customer_numbers.do', params={
+                    "customerNumbers[]": customer["asiakasnro"]
+                })
+                r.raise_for_status()
+                data = r.json()
+                for service in data:
+                    if service["tariff"].get("productgroup", "PER") != "PER":
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py:71]: I'd prefer if ... == ...: continue to reduce unnecessary indentation (the same in line 80)

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Kiertokapula Finland"
+DESCRIPTION = "Schedule for kiertokapula FI"
+URL = "https://www.kiertokapula.fi"
+TEST_CASES = {
+    # No test cases, because auth :(
+}
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py:12]: You can add test cases using 
```python
{
"bill_number": "!secret kiertonkapula_fi_bill_number"
"password": "!secret kiertonkapula_fi_bill_password"
}
```
and define these secrets in the custom_components/waste_collection_schedule/waste_collection_schedule/test/secrets.yaml file (which is not synchronized with git):

```yaml
kiertonkapula_fi_bill_number: NUMBER
kiertonkapula_fi_bill_password: PASSWORD
```

File: custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+API_URL = "https://asiakasnetti.kiertokapula.fi/kiertokapula"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Source:
+    def __init__(
+        self,
+        bill_number=None,
+        password=None,
```
Comment (5ila5) [inline custom_components/waste_collection_schedule/waste_collection_schedule/source/kiertokapula_fi.py:38]: Please do not make mandatory arguments optional

**Review by 5ila5 — CHANGES_REQUESTED** [review]
Please run the updaete_doculinks.py script to update the markdown files ([more information](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md#update-links-in-readmemd-and-infomd))

Do you think it is obvious to users of the supported cities to know to use this source, if not you can add them to the markdown overview using the EXTRA_INFO variable (for a detailed explanation see link above)

**Comment by developerfromjokela** [comment]
> Please run the updaete_doculinks.py script to update the markdown files ([more information](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md#update-links-in-readmemd-and-infomd))
> 
> 
> 
> Do you think it is obvious to users of the supported cities to know to use this source, if not you can add them to the markdown overview using the EXTRA_INFO variable (for a detailed explanation see link above)

Usually users are aware which company they're using, they're getting the bill after all. I will run the doc script 👍

**Comment by developerfromjokela** [comment]
@5ila5 I'm trying to run update_doculinks.py, but it throws an error on many sources:

```python
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

**Comment by 5ila5** [comment]
This error indicates you use a python version prior to 3.10 as this syntax was first introduced in 3.10

**Review by 5ila5 — APPROVED** [review]

**Comment by 5ila5** [comment]
Thanks for your contribuiton.

I run the script I needed to add Finnland to the country map as there was no Finland source prior to this one.
I reformatted your code based on .pre-commit-config.yaml

**Pull request merged**